resources/js/functions.js

```diff
@@ -0,0 +1,286 @@
+});
+
+$('#scaladex-search').on('blur', function(e) {
+    if (!$(e.target).is('.autocomplete-suggestion')) {
+        if (lastElementClicked != null && !lastElementClicked.is('.autocomplete-suggestion')) {
+            hideSuggestions();
+        }
+    } else {
+        hideSuggestions();
+    }
+});
+
+$('#scaladex-search').autocomplete({
+    paramName: 'q',
+    serviceUrl: 'https://index.scala-lang.org/api/autocomplete',
+    dataType: 'json',
+    beforeRender: function() {
+        showSuggestions();
+    },
+    onSearchStart: function(query) {
+        if (query == "") {
+            hideSuggestions()
+        } else {
+            showSuggestions();
+        }
+    },
+    transformResult: function(response) {
+        return {
+            suggestions: $.map(response, function(dataItem) {
+                return { value: dataItem.repository, data: 'https://scaladex.scala-lang.org/' + dataItem.organization + "/" + dataItem.repository };
+            })
+        };
+    },
+    onSearchComplete: function (query, suggestions) {
+        suggestions.length > 0 ? showSuggestions() : hideSuggestions();
+    },
+    onSelect: function (suggestion) {
+        if (suggestion.data != prevResult) {
+            prevResult = suggestion.data;
+            hideSuggestions();
+            $("#scaladex-search").blur();
+            window.open(suggestion.data, '_blank');
+        }
+    }
+
+});
+
+$(document).ready(function() {
+    $(window).on("blur", function() {
+        if ($("#scaladex-search").length) {
+            $("#scaladex-search").blur();
+            $("#scaladex-search").autocomplete().clear();
+        }
+    });
+});
+
+// TOC:
+$(document).ready(function() {
+    if ($("#sidebar-toc").length) {
+        $('#sidebar-toc').toc({exclude: 'h1, h5, h6', context: '.inner-box', autoId: true, numerate: false});
+        toggleStickyToc();
+    }
+})
+
+$(window).resize(function() {
+  toggleStickyToc();
+});
+
+var toggleStickyToc = function() {
+    if ($("#sidebar-toc").length) {
+        if ($(window).width() <= 992) {
+            $(".sidebar-toc-wrapper").unstick();
+        } else {
+            $(".sidebar-toc-wrapper").sticky({
+               topSpacing: 0,
+               bottomSpacing: 500
+            });
+        }
+    }    
+}
+
+// Blog search
+$(document).ready(function() {
+  if ($("#blog-search-bar").length) {
+    SimpleJekyllSearch({
+      searchInput: document.getElementById('blog-search-bar'),
+      resultsContainer: document.getElementById('result-container'),
+      json: '/resources/json/search.json',
+      searchResultTemplate: '<li><a href="{url}">{title}</a></li>',
+      limit: 5,      
+    });
+
+    $("#blog-search-bar").on("change paste keyup", function () {
+      if ($(this).val()) {
+        $("#result-container").show();
+      } else {
+        $("#result-container").hide();
+      }
+    });
+  }
+});
+
+// Scala in the browser
+$(document).ready(function() {
```

Comment:
Here is the javascript api to embed Scastie. I think there still some work to do to get the proper CSS
https://scastie.scala-lang.org/embedded-demo